index.js

```diff
@@ -67,20 +65,15 @@ var gs = {
     var negatives = [];
 
     globs.forEach(function(glob, index) {
-      if (typeof glob !== 'string' && !(glob instanceof RegExp)) {
+      if (typeof glob !== 'string') {
         throw new Error('Invalid glob at index ' + index);
       }
 
       var globArray = isNegative(glob) ? negatives : positives;
 
-      // create Minimatch instances for negative glob patterns
-      if (globArray === negatives && typeof glob === 'string') {
-        glob = new Minimatch(unrelative(opt.cwd, glob), opt);
-      }
-
       globArray.push({
         index: index,
-        glob: glob
+        glob: globArray === negatives ? glob.slice(1) : glob
```

Comment:
i would put isNegative in a var so it only gets called once then ternary the other two places